android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt

```diff
@@ -145,6 +153,34 @@ class HyperSdkFlutterPlugin: FlutterPlugin, MethodCallHandler, ActivityAware, Pl
         result.success(true)
     }
 
+    private fun openPaymentPage(params: Map<String, Any>?, result : Result) {
+        isHyperCheckOutLiteInteg = true
+        val invokeMethodResult = object : Result {
+            override fun success(result: Any?) {
+                Log.d(this.javaClass.canonicalName, "success: ${result.toString()}")
+                println("result = ${result.toString()}")
+            }
+
+            override fun error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
+                Log.e(this.javaClass.canonicalName, "$errorCode\n$errorMessage")
+            }
+
+            override fun notImplemented() {
+                Log.e(this.javaClass.canonicalName, "notImplemented")
+            }
+        }
+        val callback = object : HyperPaymentsCallbackAdapter() {
+            override fun onEvent(data: JSONObject, p1: JuspayResponseHandler?) {
+                try {
+                    channel.invokeMethod(data.getString("event"), data.toString(), invokeMethodResult)
+                } catch (e: Exception) {
+                    Log.e(this.javaClass.canonicalName, "Failed to invoke method from native to dart", e)
+                }
+            }
+        }
+        HyperCheckoutLite.openPaymentPage(binding!!.activity as FragmentActivity, JSONObject(params), callback)
```

Comment:
can we first add a check for whether activity is instance of FragmentActivity and throw an exception if it is not? You can reference initiate

Reply:
@harshgarg18 done